python/openassetio_traitgen/__init__.py

```diff
@@ -136,27 +135,34 @@ def generate(
     # As a convenience, log the parsed structure
     _log_package_declaration(package_declaration, logger)
 
-    if dry_run:
+    # Don't allow empty string to fall through as getattr will throw.
+    if not generator:
+        logger.error("No generator specified")
+        return
+
+    # Retrieve the generator by looking up an attribute with the
+    # requested name and generate.
+    generator_obj = getattr(generators, generator, None)
+    if generator_obj is None:
+        logger.error("Cannot find generator %s", generator)
         return
 
-    # Generate all requested languages
-    for language in languages:
+    if dry_run:
+        return
 
-        logger.info("Generating %s...", language)
-        child_logger = logger.getChild(language)
+    # Generate with requested generator
+    logger.info("Generating with %s...", generator)
+    child_logger = logger.getChild(generator)
```

Comment:
[consider] Since we only generate one language per run now, we could simplify things and get rid of the `child_logger`.

Reply:
I'm presuming this is as simple as deleting its usage here inline, let me know if not.

[97fdeee](https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/7/commits/97fdeeeca58797791a629836605c47613afb8c31)